### PR TITLE
v2.2

### PR DESCRIPTION
**Pull request by ninjarogue**

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -116,22 +159,54 @@ func parseTemplate(templateFileName string, data any) (*string, error) {
...
 }
 
+type jsonResponse struct {
+	code int
+	data []byte
+}
+
+func (j jsonResponse) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	rw.WriteHeader(j.code)
+	rw.Header().Set(ResponseHeaderContentType, ContentTypeTextHtml)
```
Comment (wellisrite) [inline response.go:169]: this should be ContentTypeTextJson ser @ninjarogue

File: response.go [hunk 2]
```diff
@@ -148,13 +223,25 @@ func (j redirectResponse) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
...
 }
 
+type xmlResponse struct {
+	code int
+	data []byte
+}
+
+func (x xmlResponse) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	rw.WriteHeader(x.code)
+	rw.Header().Set(ResponseHeaderContentType, ContentTypeTextHtml)
```
Comment (wellisrite) [inline response.go:233]: This should be `ContentTypeApplicationXml`

File: response.go [hunk 3]
```diff
@@ -116,22 +159,54 @@ func parseTemplate(templateFileName string, data any) (*string, error) {
...
-	return Json(code, map[string]any{
+	messageMap := map[string]any{
 		"message": message,
-	})
+	}
+	jsonData, err := json.Marshal(messageMap)
+	if err != nil {
+		panic(err)
+	}
+	return messageResponse{code: code, data: []byte(jsonData)}
```
Comment (wellisrite) [inline response.go:209]: please do change this too ser @ninjarogue just pass the jsonData

**Comment by Gebes** [comment]
Looks good

**Pull request merged**